### PR TITLE
Initial version of endpoint

### PR DESCRIPTION
**Pull request by shayaanjagtap**

### CODE REVIEW

File: backend/src/fun_forecast_backend/weather/dataclasses.py [hunk 1]
```diff
@@ -56,3 +56,11 @@ class WeatherDataProperties:
...
 class WeatherData:
     properties: WeatherDataProperties
+
+
+@dataclass
+class LatLongResponse:
+    lat: float
+    long: float
+    name: str
+    full_name: str
```
Comment (chelshaw) [inline backend/src/fun_forecast_backend/weather/dataclasses.py:66]: Let's make this camelCase and let the json encoder do the underscoring
```suggestion
    fullName: str
```

File: backend/src/fun_forecast_backend/weather/weather.py [hunk 2]
```diff
@@ -9,11 +9,30 @@
 
 from fun_forecast_backend.core.standard_logger import get_logger
 from fun_forecast_backend.shared.dataclasses import WeatherCondition, HourData
-from fun_forecast_backend.weather.dataclasses import WeatherApiPoint, WeatherData, Period
+from fun_forecast_backend.weather.dataclasses import WeatherApiPoint, WeatherData, Period, LatLongResponse
 
 logger = get_logger()
 
 
+def get_location_by_lat_and_long(lat: float, long: float) -> LatLongResponse:
```
Comment (chelshaw) [inline backend/src/fun_forecast_backend/weather/weather.py:17]: nit
```suggestion
def get_location_by_coords(lat: float, long: float) -> LatLongResponse:
```
Also I understand this uses `point_from_coords` which lives in weather, but maybe this *and* point_from_coords actually lives in location.py? IDK, the nature of code bends toward 🍝

File: backend/src/fun_forecast_backend/weather/weather.py [hunk 3]
```diff
@@ -9,11 +9,30 @@
...
+    try:
+        point = point_from_coords(lat, long)
+        city = point.properties.relativeLocation.properties.city.strip()
```
Comment (chelshaw) [inline backend/src/fun_forecast_backend/weather/weather.py:20]: Can we assume that every coord is going to have a `city` value? Worth testing, not sure the docs are really worth looking at 💩 but maybe

File: backend/tests/test_main.py [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
 
     async def test_get_activity_forecast_endpoint__real_input(self):
         output = await get_activity_forecast(verb="hike", lat=36.8637, long=-78.5324)
-        # self.assertTrue(isinstance(crawler, SERPApiClient))
\ No newline at end of file
+
+
+    async def test_get_location_data_by_lat_long__real_input(self):
+        output = await search_location_by_coordinates(lat=36.8637, long=-78.5324)
```
Comment (chelshaw) [inline backend/tests/test_main.py:14]: we should probably assert something right? Or is this just checking that the endpoint doesn't throw exception

**Review by chelshaw — CHANGES_REQUESTED** [review]
Couple requests, otherwise it'll do! Also could we un-gitignore the vscode workspace file b/c mine is all messed up for python

File: backend/tests/test_main.py (hunk 4 above)
Comment (shayaanjagtap) [reply backend/tests/test_main.py:14]: This was just for me to test the function without docker :) We need to mock the response and everything to make the unit test more official

File: backend/src/fun_forecast_backend/weather/weather.py (hunk 3 above)
Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/weather/weather.py:20]: Hmmm good point. Let me see what I can dig up

File: backend/src/fun_forecast_backend/weather/weather.py (hunk 2 above)
Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/weather/weather.py:17]: Yeah it doesn't have a natural fit. Definitely spaghetti. Let me see....

File: backend/src/fun_forecast_backend/weather/dataclasses.py (hunk 1 above)
Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/weather/dataclasses.py:66]: Point against that suggestion - python standards dictate underscores for variables and attributes

Comment (chelshaw) [reply backend/src/fun_forecast_backend/weather/dataclasses.py:66]: Oh then we should make them all underscores

Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/weather/dataclasses.py:66]: I see the others are all camel. I care more about consistency than anything so if this small change makes it all consistent, all g from me

**Review by chelshaw — APPROVED** [review]
🚢

**Pull request merged**